control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1396,6 +1396,22 @@ func (r *reconciler) ensureIngressControllersRemoved(ctx context.Context) (bool,
 	if len(errs) > 0 {
 		return false, fmt.Errorf("failed to delete ingress controllers: %w", errors.NewAggregate(errs))
 	}
+
+	routerPods := &corev1.PodList{}
+	if err := r.uncachedClient.List(ctx, routerPods, &client.ListOptions{Namespace: "openshift-ingress"}); err != nil {
+		return false, fmt.Errorf("failed to list pods under openshift-ingress namespace: %w", err)
+	}
+
+	for i := range routerPods.Items {
+		rp := &routerPods.Items[i]
+		if !rp.DeletionTimestamp.IsZero() {
```

Comment:
Even if the deletion timestamp is non-zero, I think you want to force delete the pod

Reply:
Sure, removed the if check.